pkg/endpoint/bpf.go

```diff
@@ -602,20 +642,35 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 			e.L3Maps.DestroyBpfMap(IPv4Egress, e.IPv4EgressMapPathLocked())
 		}
 	}
+
+	// Since the endpoint's lock will be unlocked, we need to
+	// store the current endpoint state so we can later on
+	// update the CT without requiring to lock the endpoint again.
+	isPolicyEnforced := e.IngressOrEgressIsEnforced()
```

Comment:
Doesn't this mean that the conntrack entries for an endpoint can be updated with the endpoint being unlocked? Is this desirable?